File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        
        Stack<Character> stack = new Stack<>();
        
        char[] chars = s.toCharArray();
        for(int i=0; i<chars.length; i++){
            
            if(chars[i] == '('){
                stack.push('(');
                
            }else if(stack.size() > 0 && stack.peek() == '('){
                stack.pop();
                
            }else{
                return false;
                
            }
        }

        if(stack.size() > 0){
            return false;
            
        }else{
            return true;
        }
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque

def solution(s):
    answer = True
    
    q = deque()
    for i in s:
        if i == ")":
            # 직전에 들어간 값이 "("인지 체크
            if q and q[-1] == "(":
                q.pop()
                continue
        q.append(i)
    
    if len(q) > 0:
        return False
    else:
        return True